checkov/terraform/parser.py

```diff
@@ -23,10 +25,38 @@
 _RESOURCE_REF_PATTERN = re.compile(r'[\d\w]+(\.[\d\w]+)+')
 
 
+def _filter_ignored_directories(d_names):
+    filter_ignored_directories(d_names)
+    [d_names.remove(d) for d in list(d_names) if d in [default_ml_registry.external_modules_folder_name]]
+
+
 class Parser:
     def __init__(self):
         self._parsed_directories = set()
 
+    def _init(self, directory: str, out_definitions: Optional[Dict],
+              out_evaluations_context: Dict[str, Dict[str, EvaluationContext]],
+              out_parsing_errors: Dict[str, Exception],
+              env_vars: Mapping[str, str],
+              download_external_modules: bool,
+              external_modules_download_path: str, evaluate_variables):
+        self._parsed_directories.clear()
```

Comment:
@rotemavni Why does `init()` do a `clear()`?
If you need it in `parse_directory` - do it there please